src/matching_market.sol

```diff
@@ -45,17 +47,28 @@ contract MatchingMarket is MatchingEvents, ExpiringMarket, DSNote {
     mapping(address => uint) public _dust;                      //minimum sell amount for a token to avoid dust offers
     mapping(uint => uint) public _near;         //next unsorted offer id
     uint _head;                                 //first unsorted offer id
-    uint public dustId;                         // id of the latest offer marked as dust
 
+    // dust management
+    address DAI;
+    uint256 daiDustLimit;
+    UniswapSimplePriceOracle uniswapSimplePriceOracle;
 
-    constructor(uint64 close_time) ExpiringMarket(close_time) public {
+    constructor(address _DAI, uint256 _daiDustLimit, UniswapSimplePriceOracle _uniswapSimplePriceOracle) public {
```

Comment:
If we are using an interface it would be better to declare `UniswapSimplePriceOracle`  just as `address` and then make the cast in the constructor, this way we keep a standard interface. Then in the test you can just wrap the real `UniswapSimplePriceOracle` contract with an `address()` when passing to this constructor.